### PR TITLE
Expand Settings TimePicker tap recognition to entire row

### PR DESCRIPTION
**Pull request by jlf0dev**
## Type of change
- [x] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Objective

Every option in settings allows you to tap anywhere within the row to enable the action. Every option except "Custom" timeouts. This fix pulls the option into its own data template and adds a tap recognizer on the entire row for the TimePicker.

Asana: https://app.asana.com/0/1169444489336079/1201172188894451/f

## Code changes

* **file.ext:** Description of what was changed and why
* **src/App/Pages/Settings/SettingsPage/SettingsPage.xaml:** Create new DataTemplate for Custom option in Settings. Include DataTemplate in the DataTemplateSelector
* **src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:** If TapGestureRecognizer is triggered, find the TimePicker from the StackLayout and trigger.
* **src/App/Pages/Settings/SettingsPage/SettingsPageListItemSelector.cs:** Return new DataTemplate for Custom option

## Screenshots

https://user-images.githubusercontent.com/24985544/148834516-153919be-d983-43e6-b247-2a8352507515.mov

## Testing requirements

- Test functionality of Settings options

## Before you submit
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs [hunk 1]
```diff
@@ -45,6 +46,17 @@ protected override bool OnBackButtonPressed()
             return base.OnBackButtonPressed();
         }
 
+        void ActivateTimePicker(object sender, EventArgs args)
+        {
+            var stackLayout = (ExtendedStackLayout)sender;
+            SettingsPageListItem item = (SettingsPageListItem)stackLayout.BindingContext;
+            if (item.ShowTimeInput)
+            {
+                var timePicker = stackLayout.Children.Where(x => x is TimePicker).FirstOrDefault();
```
Comment (vvolkgang) [inline src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:55]: What if we add more time pickers down the line?

File: src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs [hunk 2]
```diff
@@ -45,6 +46,17 @@ protected override bool OnBackButtonPressed()
...
+                ((TimePicker)timePicker).Focus();
```
Comment (vvolkgang) [inline src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:56]: Given that `.FirstOrDefault()`, we should probably check for null before executing this just to be safe.

**Review by vvolkgang — CHANGES_REQUESTED** [review]

File: src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs (hunk 1 above)
Comment (Hinton) [reply src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:55]: I don't think that will happen. The setting page is a bit messy as it is and we should think of ways to re-design it in the long run.

Comment (vvolkgang) [reply src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:55]: Looking forward to that as well! Until then, don't we have an ID or something we could use to pinpoint the exact item here?

Comment (jlf0dev) [reply src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:55]: There might be some confusion here. The `sender` will actually be the `ExtendedStackLayout` inside the `DataTemplate` where the `TimePicker` lives. Its not the `CollectionView` with the list of cells. This means that while there may be additional `TimePicker` cells added, the sender will always refer back to the one tapped. 

We could hardcode the `View` # inside the `ExtendedStackLayout`, but that seems like a bad idea if we ever want to reformat the cell.

**Review by vvolkgang — APPROVED** [review]

File: src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs (hunk 1 above)
Comment (vvolkgang) [reply src/App/Pages/Settings/SettingsPage/SettingsPage.xaml.cs:55]: Misunderstood, thanks for clearing it up!

**Pull request merged**